effects/mk-git-branch.nix

```
@@ -0,0 +1,58 @@
+/* Push a branchRoot to a git remote. Useful for deploying content via
+   git.
+
+   Example:
+     mkGitBranchViaEffect {
+       pushToBranch = "staging-myWebsite";
+       preGitInit = ''
+         echo "staging-myWebsite.example.com" > .domains
+       '';
+       branchRoot = "${./myWebsiteRoot}";
+       gitRemote = "git@codeberg.org";
+       hostKey = "codeberg.org ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIIVIC02vnjFyL+I4RHfvIGNtOgJMe769VTF1VR4EB3ZB";
+       owner = "myName";
+       repo = "pages";
+       committerEmail = "edolstra@gmail.com";
+       committerName = "Hercules-CI Effects";
+       authorName = "Hercules-CI Effects";
+     };
+*/
+{
+  mkEffect,
+  git,
+  openssh,
+}:
+args@{
+  gitRemote,
+  hostKey,
+  pushToBranch,
+  owner,
+  repo,
+  branchRoot,
+  committerEmail,
+  committerName,
+  authorName,
+  preGitInit ? "",
+  sshSecretName ? "ssh",
+  ...
+}:
+mkEffect {
```

The pattern is to do:

```suggestion
mkEffect (removeAttrs args ["attrs" "not" "to" "pass" "to" "mkEffect"] // {
```

Actually passing arguments through to `mkEffect` isn't bad at all, as long as you use them via the process environment. For instance, if you apply the suggestion below, you should add `"owner"` to `removeAttrs` because that will only be used through `$remote`.